NewHaven/src/VGMap.cpp

```diff
@@ -0,0 +1,261 @@
+//
+// Created by Nizz Wizz on 2020-02-05.
+//
+
+#include "VGMap.h"
+#include "boost/graph/adjacency_list.hpp"
+#include "boost/graph/graph_utility.hpp"
+#include "boost/graph/connected_components.hpp"
+// include the io stream library
+#include <iostream>
+// include the string library
+#include <string>
+#include <deque>
+using namespace std;
+using namespace boost;
+
+
+VGMap::VGMap() {
+    GenerateGraph();
+}
+// Define the deconstructor of the GameBoard Map
+VGMap::~VGMap() = default;
+// Function that goes and fetches the graph
+// generate the graph
+/*
+The gameboard will be initiated in parts, based on the configuration of the board (aka # of players)
+Each vertex (Square) will be labeled with a number. That number will be indentical to its index in the graph
+and will start at 0 at the upper left most corner of the 2 player field. From there, the vertices will be created row by row.
+It will be created in a similar way as a 2D Matrix.
+*/
+void VGMap::GenerateGraph() {
+    CreateVillageField();
+}
+// create the center 5x5 field area
+void VGMap::CreateVillageField() {
+
+    for (int vPosition = 0; vPosition < 30; vPosition++) {
+        add_vertex(*village_board);
+        (*village_board)[vPosition].setPosition(vPosition);
+        // if it isnt the first element in a row then add the previous element as a neighbour to the undirected graph
+        if (vPosition > 0 && vPosition % 5 != 0)
+            add_edge(vPosition, vPosition - 1, *village_board);
+
+
+        if (vPosition>=0 && vPosition<5) {
+            (*village_board)[vPosition].setVCost(6);
+            (*village_board)[vPosition].setRow(0);
+            (*village_board)[vPosition].setCol(vPosition % 5);
+        }
+
+        if (vPosition>=5 && vPosition<10) {
+            (*village_board)[vPosition].setVCost(5);
+            (*village_board)[vPosition].setRow(1);
+            (*village_board)[vPosition].setCol(vPosition % 5);
+        }
+
+        if (vPosition>=10 && vPosition<15) {
+            (*village_board)[vPosition].setVCost(4);
+            (*village_board)[vPosition].setRow(2);
+            (*village_board)[vPosition].setCol(vPosition % 5);
+        }
+
+        if (vPosition>=15 && vPosition<20) {
+            (*village_board)[vPosition].setVCost(3);
+            (*village_board)[vPosition].setRow(3);
+            (*village_board)[vPosition].setCol(vPosition % 5);
+        }
+        if (vPosition>=20 && vPosition<25) {
+            (*village_board)[vPosition].setVCost(2);
+            (*village_board)[vPosition].setRow(4);
+            (*village_board)[vPosition].setCol(vPosition % 5);
+        }
+
+        if (vPosition>=25 && vPosition<30) {
+            (*village_board)[vPosition].setVCost(1);
+            (*village_board)[vPosition].setRow(5);
+            (*village_board)[vPosition].setCol(vPosition % 5);
+        }
```

Comment:
Use "else if" to avoid evaluating the rest of the conditions after there has already been a match.

Reply:
done